contracts/finality/src/state/public_randomness.rs

```diff
@@ -421,19 +456,20 @@ mod tests {
         assert_eq!(
             boundary_result,
             Err(ContractError::InvalidPubRandHeight(
-                initial_commit.end_height(),
-                initial_commit.end_height()
+                pr_commit.end_height(),
+                pr_commit.end_height()
             ))
         );
 
         // === TEST CASE 3: Valid non-overlapping commitment (should pass height validation) ===
```

Comment:
Should we have any tests related to the interval?

Reply:
This test is already covering different (randomised) intervals. The main missing items imo is testing the Schnorr signature over pub rand commit. Need to be done in a separate PR due to the datagen functionalities though

Reply:
sounds good, could we track this?

Reply:
https://github.com/babylonlabs-io/rollup-bsn-contracts/issues/111